File: dynwalker/server/src/api/stats.py
import random

from sanic.response import HTTPResponse, json
from sanic.request import Request


async def get_stats(request: Request) -> HTTPResponse:
    return json({
        'description': 'Модель хождения от точки до точки.',
        'populations': {
            'species': ['walker', 'breaker'],
            'fitness': [
                {'walker': random.randint(i, i + 20), 'breaker': random.randint(80 - i, 100 - i)} for i in range(100)
            ],
        },
    })

File: dynwalker/frontend/src/redux/services/api/stats.ts
import request from '../request';

import { Species } from './area';

export interface Stats {
    description: string,
    populations: {
        species: Species[],
        fitness: { [species in Species]: number }[]
    }
}

export default {
    get(): Promise<Stats> {
        return request('stats');
    }
}
